flood_adapt/object_model/hazard/event/timeseries.py

```diff
@@ -232,129 +153,25 @@ def calculate(
-            / UnitfulTime(1, UnitTypesTime.days).convert(UnitTypesTime.seconds).value
-        )
-
-        return ts
-
+        tt_seconds = (tt - REFERENCE_TIME).total_seconds()
 
-### TIMESERIES ###
-class ITimeseries(ABC):
-    attrs: ITimeseriesModel
-
-    @abstractmethod
-    def calculate_data(self, time_step: UnitfulTime) -> np.ndarray:
-        """Interpolate timeseries data as a numpy array with the provided time step and time as index and intensity as column."""
-        ...
-
-    def to_dataframe(
-        self,
-        start_time: datetime | str,
-        end_time: datetime | str,
-        ts_start_time: UnitfulTime,
-        ts_end_time: UnitfulTime,
-        time_step: UnitfulTime,
-    ) -> pd.DataFrame:
-        """
-        Convert timeseries data to a pandas DataFrame that has time as the index and intensity as the column.
-
-        The dataframe time range is from start_time to end_time with the provided time_step.
-        The timeseries data is added to this range by first
-            - Interpolating the data to the time_step
-            - Filling the missing values with 0.
-
-        Args:
-            start_time (Union[datetime, str]): The start datetime of returned timeseries.
-                start_time is the first index of the dataframe
-            end_time (Union[datetime, str]): The end datetime of returned timeseries.
-                end_time is the last index of the dataframe (date time)
-            time_step (UnitfulTime): The time step between data points.
-
-        Note:
-            - If start_time and end_time are strings, they should be in the format DEFAULT_DATETIME_FORMAT (= "%Y-%m-%d %H:%M:%S")
-
-        Returns
-        -------
-            pd.DataFrame: A pandas DataFrame with time as the index and intensity as the column.
-            The data is interpolated to the time_step and values that fall outside of the timeseries data are filled with 0.
-        """
-        if isinstance(start_time, str):
-            start_time = datetime.strptime(start_time, DEFAULT_DATETIME_FORMAT)
-        if isinstance(end_time, str):
-            end_time = datetime.strptime(end_time, DEFAULT_DATETIME_FORMAT)
-
-        _time_step = int(time_step.convert(UnitTypesTime.seconds).value)
-
-        full_df_time_range = pd.date_range(
-            start=start_time, end=end_time, freq=f"{_time_step}S", inclusive="left"
-        )
-        full_df = pd.DataFrame(index=full_df_time_range)
-        full_df.index.name = "time"
-
-        data = self.calculate_data(time_step=time_step)
-        _time_range = pd.date_range(
-            start=(start_time + ts_start_time.to_timedelta()),
-            end=(start_time + ts_end_time.to_timedelta()),
-            inclusive="left",
-            freq=f"{_time_step}S",
+        omega = 2 * math.pi / attrs.duration.convert(UnitTypesTime.seconds)
+        phase_shift = attrs.peak_time.convert(UnitTypesTime.seconds)
+        one_period_ts = attrs.peak_value.value * np.cos(
```

Comment:
you do not need to make the timeseries for one tidal cycle first, you can simply apply the cosine to the time vector of the whole event.